sled-agent/src/sim/storage.rs

```diff
@@ -985,6 +994,15 @@ impl Storage {
                     http::StatusCode::BAD_REQUEST,
                     "Generation number too old".to_string(),
                 ));
+            } else if stored_config.generation == config.generation
```

Comment:
Should line 991 also have the sign flipped?

Reply:
Woof, yes. Great catch :facepalm: